app/routes.py

```diff
@@ -23,48 +25,60 @@
 app.route = prefix_route(app.route, ROUTE_PREFIX)
 
 
+def make_api_compliant_response(response_object: dict):
+    """
+    Making sure our responses are compliant with
+    https://github.com/geoadmin/doc-guidelines/blob/master/API.md
+    """
+    if isinstance(response_object, (Icon, IconSet)):
+        return make_response(jsonify({'success': True, **response_object.serialize()}))
```

Comment:
@ltshb I couldn't find a way to make it work without resorting to my `.serialize()` function, any idea ? (if I don't do that, the instance of the class can't be spread into the dict, and it fails)

Reply:
A pythonic way would be to have the class `Icon` and `IconSet` inherit from `collections.abc.Mapping` and then implement `__getitem__`, `__iter__`, and `__len__`. This way you won't need this check for instances except for list.

Reply:
Something like what I just pushed?

Reply:
Almost, what you pushed should work, but the `serialize()` method might be called several times (I don't know the exact process of `**` but it might call several mapping method), so with big dictionaries (lots of icons) this will have a performance issue.
Looking closer at the `Icon` and `IconSet` class, it might be difficult to implement those method in a generic way without adding performance issue, therefore you might be better of with the initial:
```python
if isinstance(response_object, (Icon, IconSet)):
        return make_response(jsonify({'success': True, **response_object.serialize()}))
```
Sorry misguidance bad idea :-/